backend/src/resolvers/mutate_posts.spec.js

```diff
@@ -0,0 +1,205 @@
+import { createTestClient } from "apollo-server-testing";
+import { gql } from "apollo-server";
+import Server from "../server";
+import Post from "../db/entities/Post";
+import { clean, close, seedPosts, seedUsers } from "../db/db";
+import driver from "../driver";
+import fixture from "../db/fixture";
+
+let mutate;
+let query;
+let server;
+beforeEach(async () => {
+  server = new Server();
+  const testClient = createTestClient(server);
+  //mock authentication
+  server.context = () => ({ id: fixture.peter.id, driver });
+
+  ({ mutate, query } = testClient);
+  await clean();
+  await seedUsers();
+});
+afterAll(async () => {
+  await clean();
+  await close();
+  driver.close();
+});
+
+describe("mutations", () => {
+  describe("WRITE_POST", () => {
+    const actionWritePost = () =>
+      mutate({
+        mutation: WRITE_POST,
+        variables: { post: { title: "Some post" } },
+      });
+    const WRITE_POST = gql`
+      mutation($post: PostInput!) {
+        write(post: $post) {
+          id
+          title
+          votes
+          author {
+            name
+          }
+        }
+      }
+    `;
+
+    it("adds a post to the database", async () => {
+      expect(await Post.all()).toHaveLength(0);
+      await actionWritePost();
+      expect(await Post.all()).toHaveLength(1);
+    });
+
+    it("responds with created post", async () => {
+      await expect(actionWritePost()).resolves.toMatchObject({
+        errors: undefined,
+        data: {
+          write: {
+            title: "Some post",
+            id: expect.any(String),
+            votes: 0,
+            author: { name: "Peter" },
```

Comment:
Here, you're not following the Arrange-Act-Assert principle. You Arrange in your beforeAll and Assert here, but you Act in another test. This is not desired. 

This is just one example.